image_transport/src/publisher.cpp

```diff
@@ -110,19 +110,36 @@ Publisher::Publisher(
   impl_->base_topic_ = image_topic;
   impl_->loader_ = loader;
 
-  std::vector<std::string> blacklist_vec;
-  std::set<std::string> blacklist;
-  // nh.getParam(impl_->base_topic_ + "/disable_pub_plugins", blacklist_vec);
-  for (size_t i = 0; i < blacklist_vec.size(); ++i) {
-    blacklist.insert(blacklist_vec[i]);
+  uint ns_len = node->get_effective_namespace().length();
+  std::string param_base_name = image_topic.substr(ns_len);
+  std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
+  if (param_base_name.front() == '.') {
+    param_base_name = param_base_name.substr(1);
   }
-
+  std::vector<std::string> whitelist_vec;
+  std::set<std::string> whitelist;
+  std::vector<std::string> all_transport_names;
   for (const auto & lookup_name : loader->getDeclaredClasses()) {
-    const std::string transport_name = erase_last_copy(lookup_name, "_pub");
-    if (blacklist.count(transport_name)) {
-      continue;
-    }
+    all_transport_names.emplace_back(erase_last_copy(lookup_name, "_pub"));
+  }
+  try {
+    whitelist_vec = node->declare_parameter<std::vector<std::string>>(
+      param_base_name + ".enable_pub_plugins", all_transport_names);
+  } catch (const rclcpp::exceptions::ParameterAlreadyDeclaredException &) {
+    RCLCPP_DEBUG_STREAM(
+      node->get_logger(), param_base_name << ".enable_pub_plugins" << " was previously declared"
+    );
+    whitelist_vec =
+      node->get_parameter(
+      param_base_name +
+      ".enable_pub_plugins").get_value<std::vector<std::string>>();
+  }
+  for (size_t i = 0; i < whitelist_vec.size(); ++i) {
```

Comment:
What is the purpose of copying the contents of the vector into a set if you're then just going to iterate over it?

Reply:
@gbiggs This is from original code, but I guess this removes duplicated plugins declared in the parameters.